interfaces.go

```diff
@@ -424,3 +424,19 @@ type StorageConstraint interface {
 	// Count is the required number of storage instances.
 	Count() uint64
 }
+
+// SLA represents the sla for the model.
```

Comment:
We have been moving interfaces out of the generic file and into the files that implement them. Please move the SLA interface into sla.go